kore/test/Test/Kore/Step/SMT/Builders.hs

```diff
@@ -0,0 +1,280 @@
+            }
+
+instance With (SentenceImport patt) Attribute where
+    s@SentenceImport {sentenceImportAttributes} `with` attribute =
+        s
+            { SentenceImport.sentenceImportAttributes =
+                sentenceImportAttributes `with` attribute
+            }
+
+instance With (SentenceSymbol level patt) Attribute where
+    s@SentenceSymbol {sentenceSymbolAttributes} `with` attribute =
+        s
+            { SentenceSymbol.sentenceSymbolAttributes =
+                sentenceSymbolAttributes `with` attribute
+            }
+
+instance With (SentenceSort level patt) Attribute where
+    s@SentenceSort {sentenceSortAttributes} `with` attribute =
+        s
+            { SentenceSort.sentenceSortAttributes =
+                sentenceSortAttributes `with` attribute
+            }
+
+instance With Attributes Attribute where
+    (Attributes attributes) `with` Attribute {getAttribute} =
+        Attributes (getAttribute:attributes)
+
+instance With (Module sentence) sentence where
+    with
+        m@Module {moduleSentences = sentences}
+        sentence
+      = m { Module.moduleSentences = sentence : sentences }
+
+indexModule
+    :: Module KoreSentence
+    -> VerifiedModule Attribute.Symbol Attribute.Axiom
+indexModule m@Module{ moduleName } =
+    indexModules moduleName [m]
+
+indexModules
+    :: ModuleName
+    -> [Module KoreSentence]
+    -> VerifiedModule Attribute.Symbol Attribute.Axiom
+indexModules moduleName modules =
+    case perhapsIndexedDefinition of
+        Left err -> (error .unlines)
+            [ "Cannot index definition:"
+            , "err = " ++ show err
+            , "modules = " ++ show modules
+            ]
+        Right indexedModules ->
+            case Map.lookup moduleName indexedModules of
+                Just indexed -> indexed
+                _ -> error
+                    "Expected to find the module in indexed definition."
+  where
+    perhapsIndexedDefinition
+        :: Either
+            (Error VerifyError)
+            (Map.Map
+                ModuleName
+                (VerifiedModule Attribute.Symbol Attribute.Axiom)
+            )
+    perhapsIndexedDefinition =
+        verifyAndIndexDefinition
+            DoNotVerifyAttributes  -- TODO: Verify attributes.
+            Builtin.koreVerifiers
+            Definition
+                { definitionAttributes = Attributes []
+                , definitionModules = modules
+                }
+
+-- TODO(virgil): either use an attribute called noJunk, or rename
+-- this constant
+noJunk :: Attribute
+noJunk = constructor
+
+constructor :: Attribute
+constructor =
+    Attribute
+    $ eraseAnnotations
+    $ patternPureToKore
+    $ mkApp (makeSort "Attribute") (makeHead "constructor") []
+
+smtlib :: Text -> Attribute
+smtlib value =
+    Attribute
+    $ eraseAnnotations
+    $ patternPureToKore
+    $ mkApp (makeSort "Attribute") (makeHead "smtlib") [mkStringLiteral value]
+
+hook :: Text -> Attribute
```

Comment:
Except for the outer `Attribute` wrapper, this is [`Kore.Attribute.Hook.hookAttribute`](https://github.com/kframework/kore/blob/master/kore/src/Kore/Attribute/Hook.hs#L44), and likewise for the other attribute definitions here.

Reply:
Fixed.